CSR signing error after upgrade to 1.16.0

### Bug Description

After upgrading istiod from 1.15.3 to 1.16.0 in a multi cluster setup the istiod pod starts up fine but after getting a request it enters a CrashLoopBackOff with the following logs:

{"level":"error","time":"2022-11-21T16:05:29.004169Z","scope":"serverca","msg":"CSR signing error (failed to find root cert from signed cert-chain (found root cert is not a ca type cert: &{[48 ... (Omitted byte array)] SHA256-RSA RSA 0xc0023ad310 3 31982574734069792384693704968739468750 CN=istio-certificate-test,O=Corp,ST=City,C=SE O= 2022-11-21 16:05:28 +0000 UTC 2022-11-22 16:05:28 +0000 UTC 5 [{2.5.29.15 true [3 2 5 160]} {2.5.29.37 false [48 20 6 8 43 6 1 5 5 7 3 1 6 8 43 6 1 5 5 7 3 2]} {2.5.29.19 true [48 0]} {2.5.29.35 false [48 ... (Omitted byte array)]} {2.5.29.17 false [48 ... (Omitted byte array)]}] [] [] [1 2] [] true false -1 false [] [30 ... (Omitted byte array)] [] [] [] [] [] [spiffe://cluster.local/ns/istio-system/sa/istio-internal-tool] false [] [] [] [] [] [] [] [] [] []}))"}
panic: interface conversion: error is *errors.errorString, not *error.Error

goroutine 968 [running]:
istio.io/istio/security/pkg/server/ca.(*Server).CreateCertificate(0xc002dd9cb0, {0x39b7828, 0xc001ace4e0}, 0xc0017f76d0)
	istio.io/istio/security/pkg/server/ca/server.go:92 +0x8e7
istio.io/api/security/v1alpha1._IstioCertificateService_CreateCertificate_Handler.func1({0x39b7828, 0xc001ace4e0}, {0x32cb760?, 0xc0017f76d0})
	istio.io/api@v0.0.0-20221109202042-b9e5d446a83d/security/v1alpha1/ca_grpc.pb.go:89 +0x78
github.com/grpc-ecosystem/go-grpc-prometheus.(*ServerMetrics).UnaryServerInterceptor.func1({0x39b7828, 0xc001ace4e0}, {0x32cb760, 0xc0017f76d0}, 0x7f84dd16b410?, 0xc002883e60)
	github.com/grpc-ecosystem/go-grpc-prometheus@v1.2.0/server_metrics.go:107 +0x87
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0x39b7828?, 0xc001ace4e0?}, {0x32cb760?, 0xc0017f76d0?})
	github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25 +0x3a
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1({0x39b7828, 0xc001ace4e0}, {0x32cb760, 0xc0017f76d0}, 0xc002ccbad0?, 0x2e73ce0?)
	github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:34 +0xbe
istio.io/api/security/v1alpha1._IstioCertificateService_CreateCertificate_Handler({0x2f5b580?, 0xc002dd9cb0}, {0x39b7828, 0xc001ace4e0}, 0xc0017cd650, 0xc002a072c0)
	istio.io/api@v0.0.0-20221109202042-b9e5d446a83d/security/v1alpha1/ca_grpc.pb.go:91 +0x138
google.golang.org/grpc.(*Server).processUnaryRPC(0xc002a52000, {0x39c54f8, 0xc003ab51e0}, 0xc0016caea0, 0xc002f048d0, 0x59ac3b0, 0x0)
	google.golang.org/grpc@v1.49.0/server.go:1301 +0xb2b
google.golang.org/grpc.(*Server).handleStream(0xc002a52000, {0x39c54f8, 0xc003ab51e0}, 0xc0016caea0, 0x0)
	google.golang.org/grpc@v1.49.0/server.go:1642 +0xa2f
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.49.0/server.go:938 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.49.0/server.go:936 +0x28a

### Version

```prose
$ istioctl version
client version: 1.15.2
control plane version: 1.16.0
data plane version: 1.15.3 (8 proxies)

$ kubectl version --short
Flag --short has been deprecated, and will be removed in the future. The --short output will become the default.
Client Version: v1.24.2
Kustomize Version: v4.5.4
Server Version: v1.23.11-gke.30
```


### Additional Information

_No response_

--- a/security/pkg/pki/ra/k8s_ra.go
+++ b/security/pkg/pki/ra/k8s_ra.go
@@ -127,7 +127,7 @@ func (r *KubernetesRA) SignWithCertChain(csrPEM []byte, certOpts ca.CertOpts) ([
 	if len(r.GetCAKeyCertBundle().GetRootCertPem()) == 0 {
 		rootCertFromCertChain, err = util.FindRootCertFromCertificateChainBytes(cert)
 		if err != nil {
-			return nil, fmt.Errorf("failed to find root cert from signed cert-chain (%v)", err.Error())
+			return nil, raerror.NewError(raerror.CSRError, fmt.Errorf("failed to find root cert from signed cert-chain (%v)", err.Error()))
 		}
 		rootCertFromMeshConfig, err = r.GetRootCertFromMeshConfig(certSigner)
 		if err != nil {
@@ -139,10 +139,10 @@ func (r *KubernetesRA) SignWithCertChain(csrPEM []byte, certOpts ca.CertOpts) ([
 			possibleRootCert = rootCertFromCertChain
 		}
 		if possibleRootCert == nil {
-			return nil, fmt.Errorf("failed to find root cert from either signed cert-chain or mesh config")
+			return nil, raerror.NewError(raerror.CSRError, fmt.Errorf("failed to find root cert from either signed cert-chain or mesh config"))
 		}
 		if verifyErr := util.VerifyCertificate(nil, cert, possibleRootCert, nil); verifyErr != nil {
-			return nil, fmt.Errorf("root cert from signed cert-chain is invalid %v ", verifyErr)
+			return nil, raerror.NewError(raerror.CSRError, fmt.Errorf("root cert from signed cert-chain is invalid %v ", verifyErr))
 		}
 		if !bytes.Equal(possibleRootCert, rootCertFromCertChain) {
 			respCertChain = append(respCertChain, string(possibleRootCert))

